Import Numpy as np
Import Panda as pd

In [1]:
import numpy as np
import pandas as pd

Import Data

In [2]:
salaries = pd.read_csv('Salary_Data.csv')

Check the Head of the Data Frame

In [3]:
salaries.head()

Age  Gender     Education Level          Job Title  Years of Experience  \
0   32    Male  Barchelor's Degree  Software Engineer                  5.0   
1   28  Female     Master's Degree       Data Analyst                  3.0   
2   45    Male                 PhD     Senior Manager                 15.0   
3   36  Female  Barchelor's Degree    Sales Associate                  7.0   
4   52    Male     Master's Degree           Director                 20.0   

     Salary  
0   90000.0  
1   65000.0  
2  150000.0  
3   60000.0  
4  200000.0

Number of Entries in the Data

In [4]:
salaries.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6702 entries, 0 to 6701
Data columns (total 6 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Age                  6702 non-null   int64  
 1   Gender               6702 non-null   object 
 2   Education Level      6701 non-null   object 
 3   Job Title            6702 non-null   object 
 4   Years of Experience  6701 non-null   float64
 5   Salary               6699 non-null   float64
dtypes: float64(2), int64(1), object(3)
memory usage: 314.3+ KB


Average Salary Given to Each Staff

In [5]:
salaries['Salary'].mean()

115326.96477086132

Minimum Salary Earned

In [6]:
salaries['Salary'].min()

350.0

Details of Lowest Paid Person

In [7]:
salaries[salaries['Salary'] == salaries['Salary'].min()]

Age Gender     Education Level                           Job Title  \
258   29   Male  Barchelor's Degree  Junior Business Operations Analyst   

     Years of Experience  Salary  
258                  1.5   350.0

Maximum Salary Earned.

In [8]:
salaries['Salary'].max()

250000.0

Details of Highest Paid Person

In [199]:
salaries[salaries['Salary'] == salaries['Salary'].max()]

Age Gender     Education Level                 Job Title  \
30     50   Male  Barchelor's Degree                       CEO   
83     52   Male                 PhD  Chief Technology Officer   
4999   45   Male  Barchelor's Degree         Financial Manager   

      Years of Experience    Salary  
30                   25.0  250000.0  
83                   24.0  250000.0  
4999                 21.0  250000.0

No of People Employed by Gender

No of Male Employed

In [201]:
salaries[salaries["Gender"] == "Male"]

Age Gender     Education Level                    Job Title  \
0      32   Male  Barchelor's Degree            Software Engineer   
2      45   Male                 PhD               Senior Manager   
4      52   Male     Master's Degree                     Director   
5      29   Male  Barchelor's Degree            Marketing Analyst   
7      31   Male  Barchelor's Degree                Sales Manager   
...   ...    ...                 ...                          ...   
6692   27   Male         High School    Digital Marketing Manager   
6694   28   Male                 PhD         Sales Representative   
6696   37   Male  Barchelor's Degree  Junior Sales Representative   
6698   32   Male         High School              Sales Associate   
6700   46   Male     Master's Degree            Marketing Manager   

      Years of Experience    Salary  
0                     5.0   90000.0  
2                    15.0  150000.0  
4                    20.0  200000.0  
5                     2.0   55000.0  
7                     4.0   80000.0  
...                   ...       ...  
6692                  2.0   40000.0  
6694                  4.0   55000.0  
6696                  6.0   75000.0  
6698                  3.0   50000.0  
6700                 14.0  140000.0  

[3674 rows x 6 columns]

No of Female Gender

In [203]:
salaries[salaries["Gender"] == "Female"]

Age  Gender     Education Level                         Job Title  \
1      28  Female     Master's Degree                      Data Analyst   
3      36  Female  Barchelor's Degree                   Sales Associate   
6      42  Female     Master's Degree                   Product Manager   
8      26  Female  Barchelor's Degree             Marketing Coordinator   
11     48  Female  Barchelor's Degree                        HR Manager   
...   ...     ...                 ...                               ...   
6693   33  Female  Barchelor's Degree         Content Marketing Manager   
6695   51  Female     Master's Degree  Senior Product Marketing Manager   
6697   49  Female                 PhD             Director of Marketing   
6699   30  Female  Barchelor's Degree                 Financial Manager   
6701   26  Female         High School                   Sales Executive   

      Years of Experience    Salary  
1                     3.0   65000.0  
3                     7.0   60000.0  
6                    12.0  120000.0  
8                     1.0   45000.0  
11                   18.0  140000.0  
...                   ...       ...  
6693                  7.0   90000.0  
6695                 19.0  190000.0  
6697                 20.0  200000.0  
6699                  4.0   55000.0  
6701                  1.0   35000.0  

[3014 rows x 6 columns]

Average Salary By Gender

In [10]:
Sal_Sex = salaries.groupby('Gender')['Salary'].mean()

In [11]:
Sal_Sex

Gender
Female    107888.998672
Male      121389.870915
Other     125869.857143
Name: Salary, dtype: float64

Highest Paid Salary by Gender



In [12]:
Max = salaries.groupby('Gender')['Salary'].max()

In [13]:
Max

Gender
Female    220000.0
Male      250000.0
Other     166109.0
Name: Salary, dtype: float64

Lowest Salary Earned By Gender

In [14]:
Min = salaries.groupby('Gender')['Salary'].min()

In [15]:
Min

Gender
Female      500.0
Male        350.0
Other     62852.0
Name: Salary, dtype: float64

Lowest Paid Staff of Undisclosed Gender

In [16]:
Males = salaries[salaries['Gender'] == 'Male']

In [17]:
Females = salaries[salaries['Gender'] == 'Female']

In [18]:
Other = salaries[salaries['Gender'] == 'Other']

In [19]:
Other

Age Gender  Education Level                 Job Title  \
2376   53  Other      High School   Senior Project Engineer   
2383   25  Other      High School       Full Stack Engineer   
2404   37  Other  Master's Degree       Full Stack Engineer   
2446   31  Other      High School        Back end Developer   
2452   23  Other      High School       Front end Developer   
2456   53  Other      High School   Senior Project Engineer   
2463   25  Other      High School       Full Stack Engineer   
2484   37  Other  Master's Degree       Full Stack Engineer   
2526   31  Other      High School        Back end Developer   
2532   23  Other      High School       Front end Developer   
2552   54  Other      High School  Senior Software Engineer   
2798   54  Other      High School  Senior Software Engineer   
2917   54  Other      High School  Senior Software Engineer   
3036   54  Other      High School  Senior Software Engineer   

      Years of Experience    Salary  
2376                 31.0  166109.0  
2383                  2.0   69032.0  
2404                 14.0  161393.0  
2446                  8.0  104127.0  
2452                  2.0   62852.0  
2456                 31.0  166109.0  
2463                  2.0   69032.0  
2484                 14.0  161393.0  
2526                  8.0  104127.0  
2532                  2.0   62852.0  
2552                 29.0  158254.0  
2798                 29.0  158966.0  
2917                 29.0  158966.0  
3036                 29.0  158966.0

In [20]:
Low_oth = Other[Other['Salary'] == Other['Salary'].min()]

In [21]:
Low_oth

Age Gender Education Level            Job Title  Years of Experience  \
2452   23  Other     High School  Front end Developer                  2.0   
2532   23  Other     High School  Front end Developer                  2.0   

       Salary  
2452  62852.0  
2532  62852.0

Highest Paid Staff of Undisclosed Gender

In [22]:
High_oth = Other[Other['Salary'] == Other['Salary'].max()]

In [23]:
High_oth

Age Gender Education Level                Job Title  \
2376   53  Other     High School  Senior Project Engineer   
2456   53  Other     High School  Senior Project Engineer   

      Years of Experience    Salary  
2376                 31.0  166109.0  
2456                 31.0  166109.0

Lowest Paid Female Staff

In [24]:
Lowest_Paid_Female = Females[Females['Salary'] == Females['Salary'].min()]

In [25]:
Lowest_Paid_Female

Age  Gender     Education Level              Job Title  \
4631   31  Female  Barchelor's Degree  Junior HR Coordinator   

      Years of Experience  Salary  
4631                  4.0   500.0

Highest Paid Female

In [26]:
Highest_Paid_Females = Females[Females['Salary'] == Females['Salary'].max()]

In [27]:
Highest_Paid_Females

Age  Gender Education Level                 Job Title  \
2258   46  Female             PhD  Director of Data Science   

      Years of Experience    Salary  
2258                 20.0  220000.0

Lowest Paid Male Staff

In [28]:
Lowest_Paid_males = Males[Males['Salary'] == Males['Salary'].min()]

In [29]:
Lowest_Paid_males

Age Gender     Education Level                           Job Title  \
258   29   Male  Barchelor's Degree  Junior Business Operations Analyst   

     Years of Experience  Salary  
258                  1.5   350.0

Average Salaries by Years of Experience

In [30]:
Sal_Exp = salaries.groupby('Years of Experience')['Salary'].mean()

In [31]:
Sal_Exp

Years of Experience
0.0      29680.233333
0.5      35000.000000
1.0      46992.846296
1.5      36279.166667
2.0      58699.457377
3.0      72944.406977
4.0      83332.090038
5.0     103111.092732
6.0     111891.146119
7.0     122108.232295
8.0     126438.138824
9.0     138021.460526
10.0    131690.322917
11.0    153060.318750
12.0    153398.064626
13.0    153002.181818
14.0    168632.363636
15.0    160664.759690
16.0    183285.438017
17.0    184053.830189
18.0    184340.412698
19.0    182430.330579
20.0    182288.721311
21.0    176734.190476
22.0    188644.531915
23.0    189573.702703
24.0    211225.421053
25.0    177803.458333
26.0    187717.285714
27.0    187922.636364
28.0    189774.812500
29.0    181437.000000
30.0    163339.833333
31.0    183027.200000
32.0    192540.800000
33.0    186400.666667
34.0    188651.000000
Name: Salary, dtype: float64

Average Salary Given Based on Educational Level

In [32]:
by_edu = salaries.groupby("Education Level").mean("Salary")

In [121]:
by_edu

Age  Years of Experience         Salary
Education Level                                                  
Barchelor's Degree  30.260668             5.419755   95082.908640
High School         26.854911             1.915179   36706.694196
Master's Degree     35.169264             9.644815  130112.056452
PhD                 41.154858            13.915267  165651.457999

Relationship between Experience and Highest Educational Level

No of PhD holders with 0 to 5 years Experience

In [34]:
PhD = salaries[(salaries["Education Level"] == 'PhD') & (salaries["Years of Experience"] >= 0) & (salaries["Years of Experience"] <= 5)]

In [35]:
PhD

Age  Gender Education Level             Job Title  Years of Experience  \
377    30    Male             PhD        Data Scientist                  5.0   
385    27    Male             PhD        Data Scientist                  2.0   
396    30  Female             PhD        Data Scientist                  5.0   
404    27    Male             PhD        Data Scientist                  2.0   
415    30  Female             PhD        Data Scientist                  5.0   
...   ...     ...             ...                   ...                  ...   
6638   28    Male             PhD  Sales Representative                  4.0   
6652   28    Male             PhD     Marketing Manager                  4.0   
6666   28    Male             PhD  Sales Representative                  4.0   
6680   28    Male             PhD     Marketing Manager                  4.0   
6694   28    Male             PhD  Sales Representative                  4.0   

        Salary  
377   180000.0  
385   115000.0  
396   180000.0  
404   115000.0  
415   180000.0  
...        ...  
6638   55000.0  
6652   55000.0  
6666   55000.0  
6680   55000.0  
6694   55000.0  

[69 rows x 6 columns]

No of PhD Holders with 6 to 10 years experience

In [36]:
PhD_1 = salaries[(salaries["Education Level"] == 'PhD') & (salaries["Years of Experience"] >= 6) & (salaries["Years of Experience"] <= 10)]

In [37]:
PhD_1

Age  Gender Education Level                         Job Title  \
9      38    Male             PhD                  Senior Scientist   
381    33    Male             PhD                    Data Scientist   
389    33    Male             PhD                    Data Scientist   
400    33    Male             PhD                    Data Scientist   
408    33    Male             PhD                    Data Scientist   
...   ...     ...             ...                               ...   
6633   36  Female             PhD                 Marketing Manager   
6647   36  Female             PhD  Senior Product Marketing Manager   
6661   36  Female             PhD                 Marketing Manager   
6675   36  Female             PhD  Senior Product Marketing Manager   
6689   36  Female             PhD                 Marketing Manager   

      Years of Experience    Salary  
9                    10.0  110000.0  
381                   8.0  190000.0  
389                   8.0  190000.0  
400                   8.0  190000.0  
408                   8.0  190000.0  
...                   ...       ...  
6633                  9.0   95000.0  
6647                 10.0  100000.0  
6661                  9.0   95000.0  
6675                 10.0  100000.0  
6689                  9.0   95000.0  

[243 rows x 6 columns]

No of PhD Holders with 11 to 15 years Experience

In [168]:
PhD_2 = salaries[(salaries["Education Level"] == 'PhD') & (salaries["Years of Experience"] >= 11) & (salaries["Years of Experience"] <= 15)]

In [39]:
PhD_2

Age  Gender Education Level                         Job Title  \
2      45    Male             PhD                    Senior Manager   
17     39    Male             PhD                   Senior Engineer   
28     43  Female             PhD                 Senior Consultant   
48     38    Male             PhD                  Senior Scientist   
90     36    Male             PhD               Principal Scientist   
...   ...     ...             ...                               ...   
6266   50    Male             PhD  Senior Product Marketing Manager   
6282   40    Male             PhD                    Sales Director   
6300   50    Male             PhD  Senior Product Marketing Manager   
6305   45  Female             PhD                 Financial Manager   
6456   42    Male             PhD         Content Marketing Manager   

      Years of Experience    Salary  
2                    15.0  150000.0  
17                   12.0  115000.0  
28                   15.0  140000.0  
48                   11.0  120000.0  
90                   11.0  120000.0  
...                   ...       ...  
6266                 15.0  150000.0  
6282                 14.0  140000.0  
6300                 15.0  150000.0  
6305                 12.0  120000.0  
6456                 15.0  150000.0  

[518 rows x 6 columns]

No of PhD Holders with 16 to 20 years Experience

In [169]:
PhD_3 = salaries[(salaries["Education Level"] == 'PhD') & (salaries["Years of Experience"] >= 16) & (salaries["Years of Experience"] <= 20)]

In [170]:
PhD_3

Age  Gender Education Level                         Job Title  \
34     46    Male             PhD                    Senior Manager   
57     43    Male             PhD                   Senior Engineer   
72     45    Male             PhD                 Research Director   
105    44    Male             PhD                Chief Data Officer   
117    48    Male             PhD                Principal Engineer   
...   ...     ...             ...                               ...   
6641   49  Female             PhD             Director of Marketing   
6655   49  Female             PhD  Senior Product Marketing Manager   
6669   49  Female             PhD             Director of Marketing   
6683   49  Female             PhD  Senior Product Marketing Manager   
6697   49  Female             PhD             Director of Marketing   

      Years of Experience    Salary  
34                   20.0  170000.0  
57                   17.0  140000.0  
72                   16.0  190000.0  
105                  16.0  220000.0  
117                  20.0  170000.0  
...                   ...       ...  
6641                 20.0  200000.0  
6655                 20.0  200000.0  
6669                 20.0  200000.0  
6683                 20.0  200000.0  
6697                 20.0  200000.0  

[435 rows x 6 columns]

No of PhD Holders with more than 20 years Experience

In [171]:
PhD_4 = salaries[(salaries["Education Level"] == 'PhD') & (salaries["Years of Experience"] >= 21)]

In [172]:
PhD_4

Age  Gender Education Level                    Job Title  \
63     47    Male             PhD        Senior Data Scientist   
83     52    Male             PhD     Chief Technology Officer   
96     47    Male             PhD           Research Scientist   
133    47    Male             PhD    Senior Research Scientist   
158    51  Female             PhD  Director of Human Resources   
...   ...     ...             ...                          ...   
4451   48    Male             PhD               Data Scientist   
4466   48    Male             PhD               Data Scientist   
4481   48    Male             PhD               Data Scientist   
4496   48    Male             PhD               Data Scientist   
4511   48    Male             PhD               Data Scientist   

      Years of Experience    Salary  
63                   21.0  180000.0  
83                   24.0  250000.0  
96                   22.0  160000.0  
133                  22.0  160000.0  
158                  23.0  185000.0  
...                   ...       ...  
4451                 22.0  219000.0  
4466                 22.0  219000.0  
4481                 22.0  219000.0  
4496                 22.0  219000.0  
4511                 22.0  219000.0  

[104 rows x 6 columns]

For Master's Degree Holders

No of Master's Degree holders with 0 to 5 years experience

In [138]:
Ma = salaries[(salaries["Education Level"] == "Master's Degree") & (salaries["Years of Experience"] >= 0) & (salaries["Years of Experience"] <= 5)]

In [139]:
Ma

Age  Gender  Education Level                  Job Title  \
1      28  Female  Master's Degree               Data Analyst   
10     29    Male  Master's Degree         Software Developer   
20     34  Female  Master's Degree           Business Analyst   
32     29    Male  Master's Degree             Data Scientist   
40     34  Female  Master's Degree                UX Designer   
...   ...     ...              ...                        ...   
6310   26    Male  Master's Degree       Sales Representative   
6326   27    Male  Master's Degree  Digital Marketing Manager   
6351   27    Male  Master's Degree  Digital Marketing Manager   
6463   27  Female  Master's Degree  Digital Marketing Manager   
6587   28  Female  Master's Degree  Digital Marketing Manager   

      Years of Experience   Salary  
1                     3.0  65000.0  
10                    3.0  75000.0  
20                    5.0  80000.0  
32                    3.0  75000.0  
40                    5.0  80000.0  
...                   ...      ...  
6310                  3.0  45000.0  
6326                  3.0  50000.0  
6351                  3.0  50000.0  
6463                  3.0  50000.0  
6587                  4.0  55000.0  

[363 rows x 6 columns]

No of Master's Degree  Holders with 6 to 10 years experience

In [56]:
Ma_1 = salaries[(salaries["Education Level"] == "Master's Degree") & (salaries["Years of Experience"] >= 6) & (salaries["Years of Experience"] <= 10)]

In [57]:
Ma1

Age  Gender  Education Level                      Job Title  \
16     33  Female  Master's Degree              Marketing Manager   
29     33    Male  Master's Degree               Product Designer   
51     33    Male  Master's Degree  Business Intelligence Analyst   
59     35    Male  Master's Degree                Product Manager   
65     38  Female  Master's Degree      Digital Marketing Manager   
...   ...     ...              ...                            ...   
6280   38    Male  Master's Degree          Director of Marketing   
6291   39  Female  Master's Degree          Director of Marketing   
6322   38    Male  Master's Degree          Director of Marketing   
6334   33  Female  Master's Degree      Content Marketing Manager   
6347   38    Male  Master's Degree          Director of Marketing   

      Years of Experience    Salary  
16                    7.0   90000.0  
29                    6.0   90000.0  
51                    7.0   85000.0  
59                    7.0  105000.0  
65                   10.0   90000.0  
...                   ...       ...  
6280                  9.0   95000.0  
6291                 10.0  100000.0  
6322                  9.0   95000.0  
6334                  6.0   75000.0  
6347                  9.0   95000.0  

[788 rows x 6 columns]

No of Master's Degree holders with 11 to 15 years Experience 

In [58]:
Ma_2 = salaries[(salaries["Education Level"] == "Master's Degree") & (salaries["Years of Experience"] >= 11) & (salaries["Years of Experience"] <= 15)]

In [59]:
Ma_2

Age  Gender  Education Level                  Job Title  \
6      42  Female  Master's Degree            Product Manager   
13     40  Female  Master's Degree            Project Manager   
24     41    Male  Master's Degree          Financial Manager   
26     37  Female  Master's Degree           Software Manager   
37     42    Male  Master's Degree           Project Engineer   
...   ...     ...              ...                        ...   
6663   43  Female  Master's Degree              Sales Manager   
6672   46    Male  Master's Degree          Marketing Manager   
6677   43  Female  Master's Degree  Digital Marketing Manager   
6691   43  Female  Master's Degree              Sales Manager   
6700   46    Male  Master's Degree          Marketing Manager   

      Years of Experience    Salary  
6                    12.0  120000.0  
13                   14.0  130000.0  
24                   13.0  140000.0  
26                   11.0  110000.0  
37                   14.0  115000.0  
...                   ...       ...  
6663                 14.0  140000.0  
6672                 14.0  140000.0  
6677                 15.0  150000.0  
6691                 14.0  140000.0  
6700                 14.0  140000.0  

[499 rows x 6 columns]

No of Master's Degree holders with 16 to 20 years experience

In [62]:
Ma_3 = salaries[(salaries["Education Level"] == "Master's Degree") & (salaries["Years of Experience"] >= 16) & (salaries["Years of Experience"] <= 20)]

In [63]:
Ma_3

Age  Gender  Education Level                         Job Title  \
4      52    Male  Master's Degree                          Director   
21     47    Male  Master's Degree                  VP of Operations   
41     48    Male  Master's Degree               Operations Director   
47     45  Female  Master's Degree             Director of Marketing   
53     47    Male  Master's Degree                     VP of Finance   
...   ...     ...              ...                               ...   
6658   46    Male  Master's Degree             Director of Marketing   
6667   51  Female  Master's Degree  Senior Product Marketing Manager   
6681   51  Female  Master's Degree         Content Marketing Manager   
6686   46    Male  Master's Degree             Director of Marketing   
6695   51  Female  Master's Degree  Senior Product Marketing Manager   

      Years of Experience    Salary  
4                    20.0  200000.0  
21                   19.0  190000.0  
41                   18.0  190000.0  
47                   16.0  180000.0  
53                   19.0  200000.0  
...                   ...       ...  
6658                 16.0  160000.0  
6667                 19.0  190000.0  
6681                 19.0  190000.0  
6686                 16.0  160000.0  
6695                 19.0  190000.0  

[131 rows x 6 columns]

No of Master's Degree Holders with more than 20 years experience

In [66]:
Ma_4 = salaries[(salaries["Education Level"] == "Master's Degree") & (salaries["Years of Experience"] >= 21)]

In [67]:
Ma_4

Age  Gender  Education Level                  Job Title  \
60     51  Female  Master's Degree     Director of Operations   
88     46    Male  Master's Degree     Senior Project Manager   
93     52  Female  Master's Degree   Senior Marketing Manager   
121    53    Male  Master's Degree      Director of Marketing   
130    50  Female  Master's Degree     Director of Operations   
...   ...     ...              ...                        ...   
3090   50  Female  Master's Degree    Senior Project Engineer   
3115   50  Female  Master's Degree   Senior Software Engineer   
3121   45  Female  Master's Degree  Software Engineer Manager   
3129   50  Female  Master's Degree   Senior Software Engineer   
4395   49    Male  Master's Degree          Marketing Manager   

      Years of Experience    Salary  
60                   23.0  170000.0  
88                   21.0  135000.0  
93                   25.0  170000.0  
121                  25.0  180000.0  
130                  21.0  160000.0  
...                   ...       ...  
3090                 22.0  182768.0  
3115                 21.0  177002.0  
3121                 22.0  183530.0  
3129                 23.0  183690.0  
4395                 23.0  228000.0  

[80 rows x 6 columns]

For High School Degree

No of People with High School Degrees with 0 to 5 years experience

In [120]:
High_School1 = salaries[(salaries["Education Level"] == 'High School') & (salaries["Years of Experience"] >= 0) & (salaries["Years of Experience"] <= 5)]

In [121]:
High_School1

Age  Gender Education Level                    Job Title  \
2383   25   Other     High School          Full Stack Engineer   
2409   24  Female     High School      Senior Project Engineer   
2434   22  Female     High School           Back end Developer   
2452   23   Other     High School          Front end Developer   
2463   25   Other     High School          Full Stack Engineer   
...   ...     ...             ...                          ...   
6684   32    Male     High School  Junior Sales Representative   
6687   26  Female     High School              Sales Associate   
6692   27    Male     High School    Digital Marketing Manager   
6698   32    Male     High School              Sales Associate   
6701   26  Female     High School              Sales Executive   

      Years of Experience   Salary  
2383                  2.0  69032.0  
2409                  1.0  52731.0  
2434                  0.0  51832.0  
2452                  2.0  62852.0  
2463                  2.0  69032.0  
...                   ...      ...  
6684                  3.0  40000.0  
6687                  1.0  35000.0  
6692                  2.0  40000.0  
6698                  3.0  50000.0  
6701                  1.0  35000.0  

[432 rows x 6 columns]

Number of People with High School Degrees with 6 to 10 years Experience

In [122]:
High_School2 = salaries[(salaries["Education Level"] == 'High School') & (salaries["Years of Experience"] > 5)  & (salaries["Years of Experience"] <= 10)]

In [123]:
High_School2

Age Gender Education Level           Job Title  Years of Experience  \
2401   28   Male     High School  Back end Developer                  7.0   
2446   31  Other     High School  Back end Developer                  8.0   
2481   28   Male     High School  Back end Developer                  7.0   
2526   31  Other     High School  Back end Developer                  8.0   

        Salary  
2401   95845.0  
2446  104127.0  
2481   95845.0  
2526  104127.0

Number of People with High School Degrees with more than 10 years Experience

In [124]:
High_School3 = salaries[(salaries["Education Level"] == 'High School') & (salaries["Years of Experience"] > 10)  & (salaries["Years of Experience"] <= 35)]

In [126]:
High_School3

Age  Gender Education Level                 Job Title  \
2376   53   Other     High School   Senior Project Engineer   
2393   30    Male     High School   Senior Project Engineer   
2418   43    Male     High School  Senior Software Engineer   
2427   49  Female     High School   Senior Project Engineer   
2456   53   Other     High School   Senior Project Engineer   
2473   30    Male     High School   Senior Project Engineer   
2498   43    Male     High School  Senior Software Engineer   
2507   49  Female     High School   Senior Project Engineer   
2552   54   Other     High School  Senior Software Engineer   
2798   54   Other     High School  Senior Software Engineer   
2917   54   Other     High School  Senior Software Engineer   
3036   54   Other     High School  Senior Software Engineer   

      Years of Experience    Salary  
2376                 31.0  166109.0  
2393                 11.0  103947.0  
2418                 22.0  165919.0  
2427                 25.0  154207.0  
2456                 31.0  166109.0  
2473                 11.0  103947.0  
2498                 22.0  165919.0  
2507                 25.0  154207.0  
2552                 29.0  158254.0  
2798                 29.0  158966.0  
2917                 29.0  158966.0  
3036                 29.0  158966.0

Number of People with Barchelor's Degree with 0 to 5 years Experience

In [142]:
B_D = salaries[(salaries["Education Level"] == "Barchelor's Degree") &  (salaries["Years of Experience"] >= 0) & (salaries["Years of Experience"] <= 5)]

In [143]:
B_D

Age  Gender     Education Level              Job Title  \
0      32    Male  Barchelor's Degree      Software Engineer   
5      29    Male  Barchelor's Degree      Marketing Analyst   
7      31    Male  Barchelor's Degree          Sales Manager   
8      26  Female  Barchelor's Degree  Marketing Coordinator   
14     27    Male  Barchelor's Degree   Customer Service Rep   
...   ...     ...                 ...                    ...   
6671   30  Female  Barchelor's Degree      Financial Manager   
6676   24    Male  Barchelor's Degree   Sales Representative   
6685   30  Female  Barchelor's Degree          Sales Manager   
6690   24    Male  Barchelor's Degree        Sales Executive   
6699   30  Female  Barchelor's Degree      Financial Manager   

      Years of Experience   Salary  
0                     5.0  90000.0  
5                     2.0  55000.0  
7                     4.0  80000.0  
8                     1.0  45000.0  
14                    2.0  40000.0  
...                   ...      ...  
6671                  4.0  55000.0  
6676                  1.0  35000.0  
6685                  5.0  70000.0  
6690                  1.0  35000.0  
6699                  4.0  55000.0  

[1942 rows x 6 columns]

Number of Barchelor's Degree Holders with 6 to 10 year Experience

In [144]:
B_D1 = salaries[(salaries["Education Level"] == "Barchelor's Degree") &  (salaries["Years of Experience"] >= 6) & (salaries["Years of Experience"] <= 10)]

In [145]:
B_D1

Age  Gender     Education Level                    Job Title  \
3      36  Female  Barchelor's Degree              Sales Associate   
12     35    Male  Barchelor's Degree            Financial Analyst   
23     36  Female  Barchelor's Degree                    Recruiter   
33     39  Female  Barchelor's Degree         Marketing Specialist   
36     35  Female  Barchelor's Degree                HR Generalist   
...   ...     ...                 ...                          ...   
6668   37    Male  Barchelor's Degree  Junior Sales Representative   
6679   33  Female  Barchelor's Degree        Director of Marketing   
6682   37    Male  Barchelor's Degree               Sales Director   
6693   33  Female  Barchelor's Degree    Content Marketing Manager   
6696   37    Male  Barchelor's Degree  Junior Sales Representative   

      Years of Experience   Salary  
3                     7.0  60000.0  
12                    6.0  65000.0  
23                    9.0  60000.0  
33                   10.0  65000.0  
36                    7.0  60000.0  
...                   ...      ...  
6668                  6.0  75000.0  
6679                  8.0  80000.0  
6682                  7.0  90000.0  
6693                  7.0  90000.0  
6696                  6.0  75000.0  

[754 rows x 6 columns]

No of Bachelor's Degree Holders with 11 to 15 years experience

In [204]:
B_D2 = salaries[(salaries["Education Level"] == "Barchelor's Degree") &  (salaries["Years of Experience"] >= 11) & (salaries["Years of Experience"] <= 15)]

In [205]:
B_D2

Age  Gender     Education Level                  Job Title  \
52     40  Female  Barchelor's Degree          Event Coordinator   
62     40  Female  Barchelor's Degree                 HR Manager   
71     39  Female  Barchelor's Degree                  Recruiter   
87     40  Female  Barchelor's Degree              HR Generalist   
95     39  Female  Barchelor's Degree        Training Specialist   
...   ...     ...                 ...                        ...   
6632   42    Male  Barchelor's Degree          Financial Manager   
6646   42    Male  Barchelor's Degree  Content Marketing Manager   
6660   42    Male  Barchelor's Degree          Financial Manager   
6674   42    Male  Barchelor's Degree  Content Marketing Manager   
6688   42    Male  Barchelor's Degree          Financial Manager   

      Years of Experience    Salary  
52                   12.0   60000.0  
62                   12.0   80000.0  
71                   11.0   70000.0  
87                   15.0   80000.0  
95                   12.0   65000.0  
...                   ...       ...  
6632                 13.0  130000.0  
6646                 13.0  130000.0  
6660                 13.0  130000.0  
6674                 13.0  130000.0  
6688                 13.0  130000.0  

[186 rows x 6 columns]

No of People with Bachelors Degree with 16 to 20 years Experience

In [164]:
B_D3 = salaries[(salaries["Education Level"] == "Barchelor's Degree") & (salaries["Years of Experience"] >= 16) & (salaries["Years of Experience"] <= 20)]

In [165]:
B_D3

Age  Gender     Education Level             Job Title  \
11     48  Female  Barchelor's Degree            HR Manager   
15     44    Male  Barchelor's Degree    Operations Manager   
66     46    Male  Barchelor's Degree            IT Manager   
92     43    Male  Barchelor's Degree  Supply Chain Manager   
99     45    Male  Barchelor's Degree    Operations Analyst   
...   ...     ...                 ...                   ...   
6212   40  Female  Barchelor's Degree     Financial Manager   
6226   40  Female  Barchelor's Degree     Financial Manager   
6240   40  Female  Barchelor's Degree     Financial Manager   
6254   40  Female  Barchelor's Degree     Financial Manager   
6336   50  Female  Barchelor's Degree         Sales Manager   

      Years of Experience    Salary  
11                   18.0  140000.0  
15                   16.0  125000.0  
66                   19.0  120000.0  
92                   18.0  105000.0  
99                   20.0  110000.0  
...                   ...       ...  
6212                 16.0  215000.0  
6226                 16.0  215000.0  
6240                 16.0  215000.0  
6254                 16.0  215000.0  
6336                 18.0  180000.0  

[90 rows x 6 columns]

No of People with Bachelors Degree with more than 20 years Experience

In [166]:
B_D4 = salaries[(salaries["Education Level"] == "Barchelor's Degree") & (salaries["Years of Experience"] >= 20)]

In [167]:
B_D4

Age  Gender     Education Level                 Job Title  \
19     51    Male  Barchelor's Degree            Sales Director   
30     50    Male  Barchelor's Degree                       CEO   
39     49    Male  Barchelor's Degree           Sales Executive   
50     51  Female  Barchelor's Degree  Customer Service Manager   
76     50  Female  Barchelor's Degree        Operations Manager   
...   ...     ...                 ...                       ...   
3072   45  Female  Barchelor's Degree   Senior Project Engineer   
3094   48  Female  Barchelor's Degree   Senior Project Engineer   
3112   43    Male  Barchelor's Degree   Senior Project Engineer   
3133   48  Female  Barchelor's Degree   Senior Project Engineer   
4999   45    Male  Barchelor's Degree         Financial Manager   

      Years of Experience    Salary  
19                   22.0  180000.0  
30                   25.0  250000.0  
39                   21.0  160000.0  
50                   22.0  130000.0  
76                   22.0  160000.0  
...                   ...       ...  
3072                 22.0  171468.0  
3094                 21.0  163209.0  
3112                 20.0  162231.0  
3133                 22.0  168691.0  
4999                 21.0  250000.0  

[63 rows x 6 columns]

No of PhD Degree Holders

In [178]:
No_of_PhD = salaries[salaries["Education Level"] == "PhD"]

In [179]:
No_of_PhD 

Age  Gender Education Level                         Job Title  \
2      45    Male             PhD                    Senior Manager   
9      38    Male             PhD                  Senior Scientist   
17     39    Male             PhD                   Senior Engineer   
28     43  Female             PhD                 Senior Consultant   
34     46    Male             PhD                    Senior Manager   
...   ...     ...             ...                               ...   
6680   28    Male             PhD                 Marketing Manager   
6683   49  Female             PhD  Senior Product Marketing Manager   
6689   36  Female             PhD                 Marketing Manager   
6694   28    Male             PhD              Sales Representative   
6697   49  Female             PhD             Director of Marketing   

      Years of Experience    Salary  
2                    15.0  150000.0  
9                    10.0  110000.0  
17                   12.0  115000.0  
28                   15.0  140000.0  
34                   20.0  170000.0  
...                   ...       ...  
6680                  4.0   55000.0  
6683                 20.0  200000.0  
6689                  9.0   95000.0  
6694                  4.0   55000.0  
6697                 20.0  200000.0  

[1369 rows x 6 columns]

No of Master's Degree Holders

In [180]:
No_of_Mas = salaries[salaries["Education Level"] == "Master's Degree"]

In [181]:
No_of_Mas 

Age  Gender  Education Level                         Job Title  \
1      28  Female  Master's Degree                      Data Analyst   
4      52    Male  Master's Degree                          Director   
6      42  Female  Master's Degree                   Product Manager   
10     29    Male  Master's Degree                Software Developer   
13     40  Female  Master's Degree                   Project Manager   
...   ...     ...              ...                               ...   
6681   51  Female  Master's Degree         Content Marketing Manager   
6686   46    Male  Master's Degree             Director of Marketing   
6691   43  Female  Master's Degree                     Sales Manager   
6695   51  Female  Master's Degree  Senior Product Marketing Manager   
6700   46    Male  Master's Degree                 Marketing Manager   

      Years of Experience    Salary  
1                     3.0   65000.0  
4                    20.0  200000.0  
6                    12.0  120000.0  
10                    3.0   75000.0  
13                   14.0  130000.0  
...                   ...       ...  
6681                 19.0  190000.0  
6686                 16.0  160000.0  
6691                 14.0  140000.0  
6695                 19.0  190000.0  
6700                 14.0  140000.0  

[1861 rows x 6 columns]

No of Barchelor's Degree Holders

In [182]:
No_of_Bar = salaries[salaries["Education Level"] == "Barchelor's Degree"]

In [183]:
No_of_Bar 

Age  Gender     Education Level                    Job Title  \
0      32    Male  Barchelor's Degree            Software Engineer   
3      36  Female  Barchelor's Degree              Sales Associate   
5      29    Male  Barchelor's Degree            Marketing Analyst   
7      31    Male  Barchelor's Degree                Sales Manager   
8      26  Female  Barchelor's Degree        Marketing Coordinator   
...   ...     ...                 ...                          ...   
6688   42    Male  Barchelor's Degree            Financial Manager   
6690   24    Male  Barchelor's Degree              Sales Executive   
6693   33  Female  Barchelor's Degree    Content Marketing Manager   
6696   37    Male  Barchelor's Degree  Junior Sales Representative   
6699   30  Female  Barchelor's Degree            Financial Manager   

      Years of Experience    Salary  
0                     5.0   90000.0  
3                     7.0   60000.0  
5                     2.0   55000.0  
7                     4.0   80000.0  
8                     1.0   45000.0  
...                   ...       ...  
6688                 13.0  130000.0  
6690                  1.0   35000.0  
6693                  7.0   90000.0  
6696                  6.0   75000.0  
6699                  4.0   55000.0  

[3023 rows x 6 columns]

In [185]:
No_of_H = salaries[salaries["Education Level"] == "High School"]

In [186]:
No_of_H

Age  Gender Education Level                    Job Title  \
2376   53   Other     High School      Senior Project Engineer   
2383   25   Other     High School          Full Stack Engineer   
2393   30    Male     High School      Senior Project Engineer   
2401   28    Male     High School           Back end Developer   
2409   24  Female     High School      Senior Project Engineer   
...   ...     ...             ...                          ...   
6684   32    Male     High School  Junior Sales Representative   
6687   26  Female     High School              Sales Associate   
6692   27    Male     High School    Digital Marketing Manager   
6698   32    Male     High School              Sales Associate   
6701   26  Female     High School              Sales Executive   

      Years of Experience    Salary  
2376                 31.0  166109.0  
2383                  2.0   69032.0  
2393                 11.0  103947.0  
2401                  7.0   95845.0  
2409                  1.0   52731.0  
...                   ...       ...  
6684                  3.0   40000.0  
6687                  1.0   35000.0  
6692                  2.0   40000.0  
6698                  3.0   50000.0  
6701                  1.0   35000.0  

[448 rows x 6 columns]

Average Age needed to acquire year of Experience

In [62]:
Age_Exp = salaries.groupby("Years of Experience")["Age"].mean()

In [63]:
Age_Exp

Years of Experience
0.0     23.291667
0.5     23.000000
1.0     25.211111
1.5     27.833333
2.0     26.560656
3.0     27.745847
4.0     28.706897
5.0     29.659148
6.0     31.186788
7.0     31.699717
8.0     33.413146
9.0     33.707895
10.0    35.682292
11.0    36.584375
12.0    37.789116
13.0    40.717703
14.0    41.245059
15.0    43.899225
16.0    43.776860
17.0    49.858491
18.0    47.190476
19.0    47.239669
20.0    46.836066
21.0    46.571429
22.0    47.851064
23.0    48.513514
24.0    49.789474
25.0    50.500000
26.0    51.285714
27.0    53.272727
28.0    53.562500
29.0    53.583333
30.0    55.333333
31.0    54.000000
32.0    54.000000
33.0    58.000000
34.0    60.000000
Name: Age, dtype: float64

Average Salaries by Age

In [64]:
Sal_Age = salaries.groupby('Age')['Salary'].mean()

In [65]:
Sal_Age

Age
21     25000.000000
22     32910.933333
23     47192.009615
24     51052.250000
25     63730.387324
26     65949.106870
27     77432.729207
28     84140.368298
29     85764.750000
30    100903.955457
31    114381.461538
32    120500.179487
33    127425.266332
34    132065.291262
35    119365.835000
36    134071.725979
37    146939.461538
38    138185.570470
39    147282.715190
40    149862.989130
41    155235.565891
42    155952.477273
43    165409.398734
44    160518.873016
45    170334.291667
46    171242.362745
47    178750.106383
48    192723.846939
49    189013.582418
50    190849.897727
51    200356.066667
52    187629.379310
53    180806.571429
54    189816.250000
55    198482.187500
56    196577.636364
57    176993.777778
58    195715.428571
60    186132.400000
61    200000.000000
62    200000.000000
Name: Salary, dtype: float64

Highest Degree obtained among age groups

In [67]:
Age_Edu = salaries[(salaries['Age'] >= 20) & (salaries['Age'] <= 25) & (salaries["Education Level"] == "Barchelor's Degree")]

In [68]:
Age_Edu

Age  Gender     Education Level                 Job Title  \
18     25  Female  Barchelor's Degree          Data Entry Clerk   
27     24    Male  Barchelor's Degree          Junior Developer   
49     25    Male  Barchelor's Degree         Help Desk Analyst   
82     25    Male  Barchelor's Degree      Sales Representative   
114    23  Female  Barchelor's Degree  Junior Marketing Analyst   
...   ...     ...                 ...                       ...   
6634   24    Male  Barchelor's Degree           Sales Executive   
6648   24    Male  Barchelor's Degree      Sales Representative   
6662   24    Male  Barchelor's Degree           Sales Executive   
6676   24    Male  Barchelor's Degree      Sales Representative   
6690   24    Male  Barchelor's Degree           Sales Executive   

      Years of Experience   Salary  
18                    0.0  35000.0  
27                    1.0  40000.0  
49                    0.0  35000.0  
82                    0.0  30000.0  
114                   0.5  35000.0  
...                   ...      ...  
6634                  1.0  35000.0  
6648                  1.0  35000.0  
6662                  1.0  35000.0  
6676                  1.0  35000.0  
6690                  1.0  35000.0  

[383 rows x 6 columns]

No of people aged Between 26 to 30 with a Barchelor's Degree

In [69]:
Age_Edu1 = salaries[(salaries['Age'] >= 26) & (salaries['Age'] <= 30) & (salaries["Education Level"] == "Barchelor's Degree")]

In [70]:
Age_Edu1

Age  Gender     Education Level                Job Title  \
5      29    Male  Barchelor's Degree        Marketing Analyst   
8      26  Female  Barchelor's Degree    Marketing Coordinator   
14     27    Male  Barchelor's Degree     Customer Service Rep   
22     30    Male  Barchelor's Degree               IT Support   
25     28  Female  Barchelor's Degree  Social Media Specialist   
...   ...     ...                 ...                      ...   
6643   30  Female  Barchelor's Degree        Financial Manager   
6657   30  Female  Barchelor's Degree            Sales Manager   
6671   30  Female  Barchelor's Degree        Financial Manager   
6685   30  Female  Barchelor's Degree            Sales Manager   
6699   30  Female  Barchelor's Degree        Financial Manager   

      Years of Experience   Salary  
5                     2.0  55000.0  
8                     1.0  45000.0  
14                    2.0  40000.0  
22                    2.0  50000.0  
25                    3.0  45000.0  
...                   ...      ...  
6643                  4.0  55000.0  
6657                  5.0  70000.0  
6671                  4.0  55000.0  
6685                  5.0  70000.0  
6699                  4.0  55000.0  

[1504 rows x 6 columns]

No of people aged between 31 and 35 years with a Barchelor's Degree

In [73]:
Age_Edu2 = salaries[(salaries['Age'] >= 31) & (salaries['Age'] <= 35) & (salaries["Education Level"] == "Barchelor's Degree")]

In [74]:
Age_Edu2

Age  Gender     Education Level                  Job Title  \
0      32    Male  Barchelor's Degree          Software Engineer   
7      31    Male  Barchelor's Degree              Sales Manager   
12     35    Male  Barchelor's Degree          Financial Analyst   
31     31  Female  Barchelor's Degree                 Accountant   
36     35  Female  Barchelor's Degree              HR Generalist   
...   ...     ...                 ...                        ...   
6637   33  Female  Barchelor's Degree  Content Marketing Manager   
6651   33  Female  Barchelor's Degree      Director of Marketing   
6665   33  Female  Barchelor's Degree  Content Marketing Manager   
6679   33  Female  Barchelor's Degree      Director of Marketing   
6693   33  Female  Barchelor's Degree  Content Marketing Manager   

      Years of Experience   Salary  
0                     5.0  90000.0  
7                     4.0  80000.0  
12                    6.0  65000.0  
31                    4.0  55000.0  
36                    7.0  60000.0  
...                   ...      ...  
6637                  7.0  90000.0  
6651                  8.0  80000.0  
6665                  7.0  90000.0  
6679                  8.0  80000.0  
6693                  7.0  90000.0  

[706 rows x 6 columns]

Number of people aged between 36 and 40 who have Barchelor's Degree

In [76]:
Age_Edu3 = salaries[(salaries['Age'] >= 31) & (salaries['Age'] <= 35) & (salaries["Education Level"] == "Barchelor's Degree")]

In [77]:
Age_Edu3

Age  Gender     Education Level                  Job Title  \
0      32    Male  Barchelor's Degree          Software Engineer   
7      31    Male  Barchelor's Degree              Sales Manager   
12     35    Male  Barchelor's Degree          Financial Analyst   
31     31  Female  Barchelor's Degree                 Accountant   
36     35  Female  Barchelor's Degree              HR Generalist   
...   ...     ...                 ...                        ...   
6637   33  Female  Barchelor's Degree  Content Marketing Manager   
6651   33  Female  Barchelor's Degree      Director of Marketing   
6665   33  Female  Barchelor's Degree  Content Marketing Manager   
6679   33  Female  Barchelor's Degree      Director of Marketing   
6693   33  Female  Barchelor's Degree  Content Marketing Manager   

      Years of Experience   Salary  
0                     5.0  90000.0  
7                     4.0  80000.0  
12                    6.0  65000.0  
31                    4.0  55000.0  
36                    7.0  60000.0  
...                   ...      ...  
6637                  7.0  90000.0  
6651                  8.0  80000.0  
6665                  7.0  90000.0  
6679                  8.0  80000.0  
6693                  7.0  90000.0  

[706 rows x 6 columns]

Number of people aged between 36 and 40 who have Barchelor's Degree

In [78]:
Age_4 = salaries[(salaries['Age'] >= 36) & (salaries['Age'] <= 40) & (salaries["Education Level"] == "Barchelor's Degree")]

In [79]:
Age_4

Age  Gender     Education Level                    Job Title  \
3      36  Female  Barchelor's Degree              Sales Associate   
23     36  Female  Barchelor's Degree                    Recruiter   
33     39  Female  Barchelor's Degree         Marketing Specialist   
43     36  Female  Barchelor's Degree     Administrative Assistant   
52     40  Female  Barchelor's Degree            Event Coordinator   
...   ...     ...                 ...                          ...   
6640   37    Male  Barchelor's Degree  Junior Sales Representative   
6654   37    Male  Barchelor's Degree               Sales Director   
6668   37    Male  Barchelor's Degree  Junior Sales Representative   
6682   37    Male  Barchelor's Degree               Sales Director   
6696   37    Male  Barchelor's Degree  Junior Sales Representative   

      Years of Experience   Salary  
3                     7.0  60000.0  
23                    9.0  60000.0  
33                   10.0  65000.0  
43                    8.0  45000.0  
52                   12.0  60000.0  
...                   ...      ...  
6640                  6.0  75000.0  
6654                  7.0  90000.0  
6668                  6.0  75000.0  
6682                  7.0  90000.0  
6696                  6.0  75000.0  

[291 rows x 6 columns]

Number Of People aged between 41 and 45 with Barchelor's Degree

In [80]:
Age_4 = salaries[(salaries['Age'] >= 36) & (salaries['Age'] <= 40) & (salaries["Education Level"] == "Barchelor's Degree")]

In [81]:
Age_4

Age  Gender     Education Level                    Job Title  \
3      36  Female  Barchelor's Degree              Sales Associate   
23     36  Female  Barchelor's Degree                    Recruiter   
33     39  Female  Barchelor's Degree         Marketing Specialist   
43     36  Female  Barchelor's Degree     Administrative Assistant   
52     40  Female  Barchelor's Degree            Event Coordinator   
...   ...     ...                 ...                          ...   
6640   37    Male  Barchelor's Degree  Junior Sales Representative   
6654   37    Male  Barchelor's Degree               Sales Director   
6668   37    Male  Barchelor's Degree  Junior Sales Representative   
6682   37    Male  Barchelor's Degree               Sales Director   
6696   37    Male  Barchelor's Degree  Junior Sales Representative   

      Years of Experience   Salary  
3                     7.0  60000.0  
23                    9.0  60000.0  
33                   10.0  65000.0  
43                    8.0  45000.0  
52                   12.0  60000.0  
...                   ...      ...  
6640                  6.0  75000.0  
6654                  7.0  90000.0  
6668                  6.0  75000.0  
6682                  7.0  90000.0  
6696                  6.0  75000.0  

[291 rows x 6 columns]

No of people Aged between 41 and 45 with  Barchelor's Degree

In [83]:
Age_5 = salaries[(salaries['Age'] >= 41) & (salaries['Age'] <= 45) & (salaries["Education Level"] == "Barchelor's Degree")]

In [84]:
Age_5

Age  Gender     Education Level                  Job Title  \
15     44    Male  Barchelor's Degree         Operations Manager   
92     43    Male  Barchelor's Degree       Supply Chain Manager   
99     45    Male  Barchelor's Degree         Operations Analyst   
120    42  Female  Barchelor's Degree              Sales Manager   
127    45    Male  Barchelor's Degree   Sales Operations Manager   
...   ...     ...                 ...                        ...   
6632   42    Male  Barchelor's Degree          Financial Manager   
6646   42    Male  Barchelor's Degree  Content Marketing Manager   
6660   42    Male  Barchelor's Degree          Financial Manager   
6674   42    Male  Barchelor's Degree  Content Marketing Manager   
6688   42    Male  Barchelor's Degree          Financial Manager   

      Years of Experience    Salary  
15                   16.0  125000.0  
92                   18.0  105000.0  
99                   20.0  110000.0  
120                  17.0  100000.0  
127                  18.0  110000.0  
...                   ...       ...  
6632                 13.0  130000.0  
6646                 13.0  130000.0  
6660                 13.0  130000.0  
6674                 13.0  130000.0  
6688                 13.0  130000.0  

[81 rows x 6 columns]

In [ ]:
No of people aged above 46 with Barchelor's Degree 

In [110]:
Age_6= salaries[(salaries['Age'] >= 46) & (salaries["Education Level"] == "Barchelor's Degree")]

In [111]:
Age_6

Age  Gender     Education Level                 Job Title  \
11     48  Female  Barchelor's Degree                HR Manager   
19     51    Male  Barchelor's Degree            Sales Director   
30     50    Male  Barchelor's Degree                       CEO   
39     49    Male  Barchelor's Degree           Sales Executive   
50     51  Female  Barchelor's Degree  Customer Service Manager   
66     46    Male  Barchelor's Degree                IT Manager   
76     50  Female  Barchelor's Degree        Operations Manager   
112    50  Female  Barchelor's Degree      Supply Chain Analyst   
143    47    Male  Barchelor's Degree      Senior Sales Manager   
2382   49  Female  Barchelor's Degree        Back end Developer   
2395   50    Male  Barchelor's Degree        Back end Developer   
2400   46  Female  Barchelor's Degree  Senior Software Engineer   
2414   52    Male  Barchelor's Degree        Back end Developer   
2421   55    Male  Barchelor's Degree   Senior Project Engineer   
2432   46    Male  Barchelor's Degree   Senior Project Engineer   
2437   49    Male  Barchelor's Degree       Full Stack Engineer   
2445   48    Male  Barchelor's Degree       Front end Developer   
2462   49  Female  Barchelor's Degree        Back end Developer   
2475   50    Male  Barchelor's Degree        Back end Developer   
2480   46  Female  Barchelor's Degree  Senior Software Engineer   
2494   52    Male  Barchelor's Degree        Back end Developer   
2501   55    Male  Barchelor's Degree   Senior Project Engineer   
2512   46    Male  Barchelor's Degree   Senior Project Engineer   
2517   49    Male  Barchelor's Degree       Full Stack Engineer   
2525   48    Male  Barchelor's Degree       Front end Developer   
2541   54    Male  Barchelor's Degree   Senior Project Engineer   
2562   46  Female  Barchelor's Degree        Back end Developer   
2570   46    Male  Barchelor's Degree   Senior Project Engineer   
2575   50  Female  Barchelor's Degree   Senior Project Engineer   
2581   54    Male  Barchelor's Degree       Full Stack Engineer   
2605   55  Female  Barchelor's Degree   Senior Project Engineer   
2627   49    Male  Barchelor's Degree        Back end Developer   
2631   46    Male  Barchelor's Degree  Senior Software Engineer   
2641   54    Male  Barchelor's Degree  Senior Software Engineer   
2650   54    Male  Barchelor's Degree       Full Stack Engineer   
2787   54    Male  Barchelor's Degree   Senior Project Engineer   
2808   46  Female  Barchelor's Degree        Back end Developer   
2816   46    Male  Barchelor's Degree   Senior Project Engineer   
2821   50  Female  Barchelor's Degree   Senior Project Engineer   
2827   54    Male  Barchelor's Degree       Full Stack Engineer   
2848   46  Female  Barchelor's Degree        Back end Developer   
2873   46  Female  Barchelor's Degree        Back end Developer   
2906   54    Male  Barchelor's Degree   Senior Project Engineer   
2927   46  Female  Barchelor's Degree        Back end Developer   
2935   46    Male  Barchelor's Degree   Senior Project Engineer   
2940   50  Female  Barchelor's Degree   Senior Project Engineer   
2946   54    Male  Barchelor's Degree       Full Stack Engineer   
2967   46  Female  Barchelor's Degree        Back end Developer   
2992   46  Female  Barchelor's Degree        Back end Developer   
3025   54    Male  Barchelor's Degree   Senior Project Engineer   
3046   46  Female  Barchelor's Degree        Back end Developer   
3054   46    Male  Barchelor's Degree   Senior Project Engineer   
3059   50  Female  Barchelor's Degree   Senior Project Engineer   
3065   54    Male  Barchelor's Degree       Full Stack Engineer   
3094   48  Female  Barchelor's Degree   Senior Project Engineer   
3133   48  Female  Barchelor's Degree   Senior Project Engineer   
6315   50  Female  Barchelor's Degree             Sales Manager   
6336   50  Female  Barchelor's Degree             Sales Manager   

      Years of Experience    Salary  
11            

Checking the various age groups with High School Degree as their highest qualification

No of people with aged between 20 and 25 years with High School degree as their highest Qualification

In [187]:
Age_H =  salaries[(salaries['Age'] >= 20) & (salaries['Age'] <= 25) & (salaries["Education Level"] == "High School")]

In [188]:
Age_H

Age  Gender Education Level                Job Title  \
2383   25   Other     High School      Full Stack Engineer   
2409   24  Female     High School  Senior Project Engineer   
2434   22  Female     High School       Back end Developer   
2452   23   Other     High School      Front end Developer   
2463   25   Other     High School      Full Stack Engineer   
...   ...     ...             ...                      ...   
6535   25  Female     High School          Sales Executive   
6549   25  Female     High School          Sales Executive   
6563   25  Female     High School          Sales Executive   
6577   25  Female     High School          Sales Executive   
6584   25    Male     High School          Sales Associate   

      Years of Experience   Salary  
2383                  2.0  69032.0  
2409                  1.0  52731.0  
2434                  0.0  51832.0  
2452                  2.0  62852.0  
2463                  2.0  69032.0  
...                   ...      ...  
6535                  1.0  35000.0  
6549                  1.0  35000.0  
6563                  1.0  35000.0  
6577                  1.0  35000.0  
6584                  1.0  35000.0  

[201 rows x 6 columns]

People aged between 26 & 30 with High School Degree as their highest Degree

In [80]:
Age_H1 = salaries[(salaries['Age'] >= 26) & (salaries['Age'] <= 30) & (salaries["Education Level"] == "High School")]

In [81]:
Age_H1

Age  Gender Education Level                  Job Title  \
2393   30    Male     High School    Senior Project Engineer   
2401   28    Male     High School         Back end Developer   
2473   30    Male     High School    Senior Project Engineer   
2481   28    Male     High School         Back end Developer   
2574   27    Male     High School         Back end Developer   
...   ...     ...             ...                        ...   
6673   26  Female     High School            Sales Executive   
6678   27    Male     High School              Sales Manager   
6687   26  Female     High School            Sales Associate   
6692   27    Male     High School  Digital Marketing Manager   
6701   26  Female     High School            Sales Executive   

      Years of Experience    Salary  
2393                 11.0  103947.0  
2401                  7.0   95845.0  
2473                 11.0  103947.0  
2481                  7.0   95845.0  
2574                  4.0   79767.0  
...                   ...       ...  
6673                  1.0   35000.0  
6678                  2.0   40000.0  
6687                  1.0   35000.0  
6692                  2.0   40000.0  
6701                  1.0   35000.0  

[184 rows x 6 columns]

People aged between 31 and 35 with High School Degree as their Highest Degree

In [90]:
Age_H2 = salaries[(salaries['Age'] >= 31) & (salaries['Age'] <= 35) & (salaries["Education Level"] == "High School")]

In [91]:
Age_H2

Age  Gender Education Level                    Job Title  \
2446   31   Other     High School           Back end Developer   
2526   31   Other     High School           Back end Developer   
3395   31    Male     High School         Junior Web Developer   
3405   31  Female     High School        Junior HR Coordinator   
3425   31    Male     High School         Junior HR Generalist   
3433   33    Male     High School        Junior HR Coordinator   
3444   31    Male     High School         Junior HR Generalist   
3463   31    Male     High School         Junior HR Generalist   
3482   31    Male     High School         Junior HR Generalist   
3501   31    Male     High School         Junior HR Generalist   
3524   31  Female     High School        Junior HR Coordinator   
3544   31    Male     High School         Junior HR Generalist   
3552   33    Male     High School        Junior HR Coordinator   
3563   31    Male     High School         Junior HR Generalist   
3582   31    Male     High School         Junior HR Generalist   
3601   31    Male     High School         Junior HR Generalist   
3620   31    Male     High School         Junior HR Generalist   
4391   33    Male     High School       Junior Sales Associate   
4409   31  Female     High School       Junior Sales Associate   
4424   31  Female     High School       Junior Sales Associate   
4439   31  Female     High School       Junior Sales Associate   
4454   31  Female     High School       Junior Sales Associate   
4469   31  Female     High School       Junior Sales Associate   
4484   31  Female     High School       Junior Sales Associate   
4499   31  Female     High School       Junior Sales Associate   
6271   31  Female     High School            Financial Manager   
6303   31  Female     High School              Sales Associate   
6346   31  Female     High School  Junior Sales Representative   
6357   31    Male     High School  Junior Sales Representative   
6371   31    Male     High School              Sales Associate   
6385   31    Male     High School              Sales Associate   
6399   31    Male     High School              Sales Associate   
6413   31    Male     High School              Sales Associate   
6427   31    Male     High School              Sales Associate   
6441   31    Male     High School              Sales Associate   
6457   33  Female     High School              Sales Executive   
6476   31    Male     High School  Junior Sales Representative   
6490   31    Male     High School              Sales Associate   
6504   31    Male     High School              Sales Associate   
6518   31    Male     High School              Sales Associate   
6532   31    Male     High School              Sales Associate   
6546   31    Male     High School              Sales Associate   
6560   31    Male     High School              Sales Associate   
6574   31    Male     High School              Sales Associate   
6581   34  Female     High School              Sales Executive   
6600   32    Male     High School  Junior Sales Representative   
6614   32    Male     High School              Sales Associate   
6628   32    Male     High School  Junior Sales Representative   
6642   32    Male     High School              Sales Associate   
6656   32    Male     High School  Junior Sales Representative   
6670   32    Male     High School              Sales Associate   
6684   32    Male     High School  Junior Sales Representative   
6698   32    Male     High School              Sales Associate   

      Years of Experience    Salary  
2446                  8.0  104127.0  
2526                  8.0  104127.0  
3395                  4.0   55000.0  
3405                  3.0   38000.0  
3425                  2.0   35000.0  
3433                  3.0   38000.0  
3444                  2.0   35000.0  
3463                  2.0   35000.0  
3482                  2.0   35000.0  
3501                  2.0   35000.0  
3524                  3

People aged above 40 with High School Degree as their Highest Degree

In [94]:
Age_H3 = salaries[(salaries['Age'] >= 36) & (salaries["Education Level"] == "High School")]

In [95]:
Age_H3

Age  Gender Education Level                 Job Title  \
2376   53   Other     High School   Senior Project Engineer   
2418   43    Male     High School  Senior Software Engineer   
2427   49  Female     High School   Senior Project Engineer   
2456   53   Other     High School   Senior Project Engineer   
2498   43    Male     High School  Senior Software Engineer   
2507   49  Female     High School   Senior Project Engineer   
2552   54   Other     High School  Senior Software Engineer   
2798   54   Other     High School  Senior Software Engineer   
2917   54   Other     High School  Senior Software Engineer   
3036   54   Other     High School  Senior Software Engineer   

      Years of Experience    Salary  
2376                 31.0  166109.0  
2418                 22.0  165919.0  
2427                 25.0  154207.0  
2456                 31.0  166109.0  
2498                 22.0  165919.0  
2507                 25.0  154207.0  
2552                 29.0  158254.0  
2798                 29.0  158966.0  
2917                 29.0  158966.0  
3036                 29.0  158966.0

Checking the various age groups with Master's Degree as their highest qualification

Number of people aged between 20 and 25 years

In [90]:
Age_M = salaries[(salaries['Age'] >= 20) & (salaries['Age'] <= 25) & (salaries["Education Level"] == "Master's Degree")]

In [189]:
Age_M

Age  Gender Education Level             Job Title  Years of Experience  \
377    30    Male             PhD        Data Scientist                  5.0   
385    27    Male             PhD        Data Scientist                  2.0   
396    30  Female             PhD        Data Scientist                  5.0   
404    27    Male             PhD        Data Scientist                  2.0   
415    30  Female             PhD        Data Scientist                  5.0   
...   ...     ...             ...                   ...                  ...   
6638   28    Male             PhD  Sales Representative                  4.0   
6652   28    Male             PhD     Marketing Manager                  4.0   
6666   28    Male             PhD  Sales Representative                  4.0   
6680   28    Male             PhD     Marketing Manager                  4.0   
6694   28    Male             PhD  Sales Representative                  4.0   

        Salary  
377   180000.0  
385   115000.0  
396   180000.0  
404   115000.0  
415   180000.0  
...        ...  
6638   55000.0  
6652   55000.0  
6666   55000.0  
6680   55000.0  
6694   55000.0  

[123 rows x 6 columns]

No of People Aged between 26 and 30 with Master's Degree as their Highest Qualification

In [95]:
Age_M1 = salaries[(salaries['Age'] >= 26) & (salaries['Age'] <= 30) & (salaries["Education Level"] == "Master's Degree")]

In [97]:
Age_M1

Age  Gender  Education Level                  Job Title  \
1      28  Female  Master's Degree               Data Analyst   
10     29    Male  Master's Degree         Software Developer   
32     29    Male  Master's Degree             Data Scientist   
56     27  Female  Master's Degree              UX Researcher   
173    26  Female  Master's Degree      Junior Data Scientist   
...   ...     ...              ...                        ...   
6310   26    Male  Master's Degree       Sales Representative   
6326   27    Male  Master's Degree  Digital Marketing Manager   
6351   27    Male  Master's Degree  Digital Marketing Manager   
6463   27  Female  Master's Degree  Digital Marketing Manager   
6587   28  Female  Master's Degree  Digital Marketing Manager   

      Years of Experience   Salary  
1                     3.0  65000.0  
10                    3.0  75000.0  
32                    3.0  75000.0  
56                    2.0  65000.0  
173                   1.5  45000.0  
...                   ...      ...  
6310                  3.0  45000.0  
6326                  3.0  50000.0  
6351                  3.0  50000.0  
6463                  3.0  50000.0  
6587                  4.0  55000.0  

[421 rows x 6 columns]

No of People aged between 31 and 35 who have Master's Degree as their Highest Qualification

In [99]:
Age_M2 = salaries[(salaries['Age'] >= 31) & (salaries['Age'] <= 35) & (salaries["Education Level"] == "Master's Degree")]

In [100]:
Age_M2

Age  Gender  Education Level                      Job Title  \
16     33  Female  Master's Degree              Marketing Manager   
20     34  Female  Master's Degree               Business Analyst   
29     33    Male  Master's Degree               Product Designer   
40     34  Female  Master's Degree                    UX Designer   
51     33    Male  Master's Degree  Business Intelligence Analyst   
...   ...     ...              ...                            ...   
6230   33  Female  Master's Degree              Marketing Manager   
6244   33  Female  Master's Degree              Marketing Manager   
6264   33    Male  Master's Degree      Content Marketing Manager   
6298   32    Male  Master's Degree      Content Marketing Manager   
6334   33  Female  Master's Degree      Content Marketing Manager   

      Years of Experience    Salary  
16                    7.0   90000.0  
20                    5.0   80000.0  
29                    6.0   90000.0  
40                    5.0   80000.0  
51                    7.0   85000.0  
...                   ...       ...  
6230                  9.0  135000.0  
6244                  9.0  135000.0  
6264                  5.0   70000.0  
6298                  5.0   70000.0  
6334                  6.0   75000.0  

[613 rows x 6 columns]

Number of People Aged between 36 to 40 with Master's Degree as their Highest Qualification

In [190]:
Age_M3 = salaries[(salaries['Age'] >= 36) & (salaries['Age'] <= 40) & (salaries["Education Level"] == "Master's Degree")]

In [191]:
Age_M3

Age  Gender  Education Level                   Job Title  \
13     40  Female  Master's Degree             Project Manager   
26     37  Female  Master's Degree            Software Manager   
65     38  Female  Master's Degree   Digital Marketing Manager   
98     38  Female  Master's Degree    Public Relations Manager   
104    39  Female  Master's Degree  Senior Project Coordinator   
...   ...     ...              ...                         ...   
6291   39  Female  Master's Degree       Director of Marketing   
6322   38    Male  Master's Degree       Director of Marketing   
6347   38    Male  Master's Degree       Director of Marketing   
6454   37    Male  Master's Degree           Marketing Manager   
6578   38    Male  Master's Degree           Marketing Manager   

      Years of Experience    Salary  
13                   14.0  130000.0  
26                   11.0  110000.0  
65                   10.0   90000.0  
98                   10.0   90000.0  
104                  13.0   80000.0  
...                   ...       ...  
6291                 10.0  100000.0  
6322                  9.0   95000.0  
6347                  9.0   95000.0  
6454                 11.0  110000.0  
6578                 12.0  120000.0  

[367 rows x 6 columns]

No of people between 41 and 45 with Master's Degree as their Highest Degree

In [192]:
Age_M4= salaries[(salaries['Age'] >= 41) & (salaries['Age'] <= 45) & (salaries["Education Level"] == "Master's Degree")]

In [193]:
Age_M4

Age  Gender  Education Level                  Job Title  \
6      42  Female  Master's Degree            Product Manager   
24     41    Male  Master's Degree          Financial Manager   
37     42    Male  Master's Degree           Project Engineer   
44     41    Male  Master's Degree        Strategy Consultant   
47     45  Female  Master's Degree      Director of Marketing   
...   ...     ...              ...                        ...   
6635   43  Female  Master's Degree              Sales Manager   
6649   43  Female  Master's Degree  Digital Marketing Manager   
6663   43  Female  Master's Degree              Sales Manager   
6677   43  Female  Master's Degree  Digital Marketing Manager   
6691   43  Female  Master's Degree              Sales Manager   

      Years of Experience    Salary  
6                    12.0  120000.0  
24                   13.0  140000.0  
37                   14.0  115000.0  
44                   13.0  130000.0  
47                   16.0  180000.0  
...                   ...       ...  
6635                 14.0  140000.0  
6649                 15.0  150000.0  
6663                 14.0  140000.0  
6677                 15.0  150000.0  
6691                 14.0  140000.0  

[280 rows x 6 columns]

People aged above 46 with Master's Degree as their Highest Degree

In [194]:
Age_M5 = salaries[(salaries['Age'] >= 46) & (salaries["Education Level"] == "Master's Degree")]

In [195]:
Age_M5

Age  Gender  Education Level                         Job Title  \
4      52    Male  Master's Degree                          Director   
21     47    Male  Master's Degree                  VP of Operations   
41     48    Male  Master's Degree               Operations Director   
53     47    Male  Master's Degree                     VP of Finance   
60     51  Female  Master's Degree            Director of Operations   
...   ...     ...              ...                               ...   
6672   46    Male  Master's Degree                 Marketing Manager   
6681   51  Female  Master's Degree         Content Marketing Manager   
6686   46    Male  Master's Degree             Director of Marketing   
6695   51  Female  Master's Degree  Senior Product Marketing Manager   
6700   46    Male  Master's Degree                 Marketing Manager   

      Years of Experience    Salary  
4                    20.0  200000.0  
21                   19.0  190000.0  
41                   18.0  190000.0  
53                   19.0  200000.0  
60                   23.0  170000.0  
...                   ...       ...  
6672                 14.0  140000.0  
6681                 19.0  190000.0  
6686                 16.0  160000.0  
6695                 19.0  190000.0  
6700                 14.0  140000.0  

[121 rows x 6 columns]

People aged between 20 and 25 with PhD as their Highest Degree

In [100]:
Age_P = salaries[(salaries['Age'] >= 20) & (salaries['Age'] <= 25) & (salaries["Education Level"] == "PhD")]

In [101]:
Age_P

Age Gender Education Level                  Job Title  \
2374   25   Male             PhD  Software Engineer Manager   
2412   23   Male             PhD  Software Engineer Manager   
2454   25   Male             PhD  Software Engineer Manager   
2492   23   Male             PhD  Software Engineer Manager   
2560   24   Male             PhD  Software Engineer Manager   
2583   23   Male             PhD  Software Engineer Manager   
2652   23   Male             PhD  Software Engineer Manager   
2806   24   Male             PhD  Software Engineer Manager   
2829   23   Male             PhD  Software Engineer Manager   
2854   23   Male             PhD  Software Engineer Manager   
2879   23   Male             PhD  Software Engineer Manager   
2925   24   Male             PhD  Software Engineer Manager   
2948   23   Male             PhD  Software Engineer Manager   
2973   23   Male             PhD  Software Engineer Manager   
2998   23   Male             PhD  Software Engineer Manager   
3044   24   Male             PhD  Software Engineer Manager   
3067   23   Male             PhD  Software Engineer Manager   
6333   24   Male             PhD  Digital Marketing Manager   

      Years of Experience    Salary  
2374                 12.0  169159.0  
2412                  1.0  119836.0  
2454                 12.0  169159.0  
2492                  1.0  119836.0  
2560                  1.0  106132.0  
2583                  0.0   52612.0  
2652                  1.0     579.0  
2806                  1.0  106686.0  
2829                  0.0   52612.0  
2854                  0.0   52779.0  
2879                  0.0   52779.0  
2925                  1.0  106686.0  
2948                  0.0   52612.0  
2973                  0.0   52779.0  
2998                  0.0   52779.0  
3044                  1.0  106686.0  
3067                  0.0   52612.0  
6333                  1.0   30000.0

People aged between 26 and 30 with PhD as their Highest Degree

In [103]:
Age_P1 = salaries[(salaries['Age'] >= 26) & (salaries['Age'] <= 30) & (salaries["Education Level"] == "PhD")]

In [104]:
Age_P1

Age  Gender Education Level             Job Title  Years of Experience  \
377    30    Male             PhD        Data Scientist                  5.0   
385    27    Male             PhD        Data Scientist                  2.0   
396    30  Female             PhD        Data Scientist                  5.0   
404    27    Male             PhD        Data Scientist                  2.0   
415    30  Female             PhD        Data Scientist                  5.0   
...   ...     ...             ...                   ...                  ...   
6638   28    Male             PhD  Sales Representative                  4.0   
6652   28    Male             PhD     Marketing Manager                  4.0   
6666   28    Male             PhD  Sales Representative                  4.0   
6680   28    Male             PhD     Marketing Manager                  4.0   
6694   28    Male             PhD  Sales Representative                  4.0   

        Salary  
377   180000.0  
385   115000.0  
396   180000.0  
404   115000.0  
415   180000.0  
...        ...  
6638   55000.0  
6652   55000.0  
6666   55000.0  
6680   55000.0  
6694   55000.0  

[123 rows x 6 columns]

People Aged Between 31 and 35 with PhD as their Highest Degree

In [107]:
Age_P2 = salaries[(salaries['Age'] >= 31) & (salaries['Age'] <= 35) & (salaries["Education Level"] == "PhD")]

In [108]:
Age_P2

Age  Gender Education Level                         Job Title  \
381    33    Male             PhD                    Data Scientist   
389    33    Male             PhD                    Data Scientist   
400    33    Male             PhD                    Data Scientist   
408    33    Male             PhD                    Data Scientist   
419    33    Male             PhD                    Data Scientist   
...   ...     ...             ...                               ...   
6523   35  Female             PhD  Senior Product Marketing Manager   
6537   35  Female             PhD  Senior Product Marketing Manager   
6551   35  Female             PhD  Senior Product Marketing Manager   
6565   35  Female             PhD  Senior Product Marketing Manager   
6586   32    Male             PhD  Senior Product Marketing Manager   

      Years of Experience    Salary  
381                   8.0  190000.0  
389                   8.0  190000.0  
400                   8.0  190000.0  
408                   8.0  190000.0  
419                   8.0  190000.0  
...                   ...       ...  
6523                  9.0   95000.0  
6537                  9.0   95000.0  
6551                  9.0   95000.0  
6565                  9.0   95000.0  
6586                  8.0   80000.0  

[251 rows x 6 columns]

No of people Aged Between 36 and 40 with PhD as their Highest Degree 

In [111]:
Age_P3 = salaries[(salaries['Age'] >= 36) & (salaries['Age'] <= 40) & (salaries["Education Level"] == "PhD")]

In [112]:
Age_P3

Age  Gender Education Level                         Job Title  \
9      38    Male             PhD                  Senior Scientist   
17     39    Male             PhD                   Senior Engineer   
48     38    Male             PhD                  Senior Scientist   
90     36    Male             PhD               Principal Scientist   
1101   37    Male             PhD                   Product Manager   
...   ...     ...             ...                               ...   
6633   36  Female             PhD                 Marketing Manager   
6647   36  Female             PhD  Senior Product Marketing Manager   
6661   36  Female             PhD                 Marketing Manager   
6675   36  Female             PhD  Senior Product Marketing Manager   
6689   36  Female             PhD                 Marketing Manager   

      Years of Experience    Salary  
9                    10.0  110000.0  
17                   12.0  115000.0  
48                   11.0  120000.0  
90                   11.0  120000.0  
1101                 14.0  195000.0  
...                   ...       ...  
6633                  9.0   95000.0  
6647                 10.0  100000.0  
6661                  9.0   95000.0  
6675                 10.0  100000.0  
6689                  9.0   95000.0  

[179 rows x 6 columns]

No of People aged between 41 and 45 with PhD as their Highest Degree 

In [117]:
Age_P4 = salaries[(salaries['Age'] >= 41) & (salaries['Age'] <= 45) & (salaries["Education Level"] == "PhD")]

In [196]:
Age_P4

Age  Gender Education Level                  Job Title  \
2      45    Male             PhD             Senior Manager   
28     43  Female             PhD          Senior Consultant   
57     43    Male             PhD            Senior Engineer   
72     45    Male             PhD          Research Director   
105    44    Male             PhD         Chief Data Officer   
...   ...     ...             ...                        ...   
4576   42    Male             PhD             Director of HR   
4628   42    Male             PhD             Director of HR   
6305   45  Female             PhD          Financial Manager   
6456   42    Male             PhD  Content Marketing Manager   
6580   43    Male             PhD  Content Marketing Manager   

      Years of Experience    Salary  
2                    15.0  150000.0  
28                   15.0  140000.0  
57                   17.0  140000.0  
72                   16.0  190000.0  
105                  16.0  220000.0  
...                   ...       ...  
4576                 14.0  130000.0  
4628                 14.0  130000.0  
6305                 12.0  120000.0  
6456                 15.0  150000.0  
6580                 16.0  160000.0  

[370 rows x 6 columns]

No of People Aged above 45 with PhD Degrees as their Highest Degree

In [197]:
Age_P5 = salaries[(salaries['Age'] >= 46) & (salaries["Education Level"] == "PhD")]

In [198]:
Age_P5

Age  Gender Education Level                         Job Title  \
34     46    Male             PhD                    Senior Manager   
63     47    Male             PhD             Senior Data Scientist   
83     52    Male             PhD          Chief Technology Officer   
96     47    Male             PhD                Research Scientist   
117    48    Male             PhD                Principal Engineer   
...   ...     ...             ...                               ...   
6641   49  Female             PhD             Director of Marketing   
6655   49  Female             PhD  Senior Product Marketing Manager   
6669   49  Female             PhD             Director of Marketing   
6683   49  Female             PhD  Senior Product Marketing Manager   
6697   49  Female             PhD             Director of Marketing   

      Years of Experience    Salary  
34                   20.0  170000.0  
63                   21.0  180000.0  
83                   24.0  250000.0  
96                   22.0  160000.0  
117                  20.0  170000.0  
...                   ...       ...  
6641                 20.0  200000.0  
6655                 20.0  200000.0  
6669                 20.0  200000.0  
6683                 20.0  200000.0  
6697                 20.0  200000.0  

[428 rows x 6 columns]

Analysis: 

From the Data, we see that Majority of the Employees in the Company have Barchelor's Degree holder as their highest degree with 3023 people which is about 45 percent of the total number of Employees. Followed by Master's Degree Holders who are 1861 people (about 27.7%) in Number with the least employed qualification being High School Degree (6.68%) and PhD Degree holders accounting for 20.62%. Among the Employees, three Genders were Represented (Male, Female and Others). Among the Genders represented Male were the most employed, making up about 54.8 percent (3674 out of 6702) of the employees while Women occupy about 45% with 0.2 percent identifying as Others.

The Highest Paid Salary was 250,000 and the lowest Paid Salary is 350 and the Average Salary given to the staffs in the company is 115326.9. The Highest Paid Male Employees were 3 in number and they were all earning 250,000 while the highest Paid Female was paid 220,000 and the highest paid Non Binary (Others) were paid 166,109. The Lowest Salary earned by the male Gender is 350 while that of the female Gender is  500 while that of those who identify as others is 62,852. However when we we check the average salary across each Gender, we see that those who Identify as others edge it slightly as they earn about 125869.86 compared to 121389.87 and  107888.99 for the male and Female Genders respectively. 

The Average Salary by Educational Level as shown in Output 121 above tells us that salary increases with educational Level. Also, we see a positive correlation between Years of Experience and the Educatonal Level of the Employees (i.e Those with Higher Educational Qualification have more years of Experience than those with Lower Educational Level/Qualification). We see that 89% (2,696 out of 3,023) of Barchelor's Degree Holders have Years of Experience between 0 - 10 years and 97.32 percent (436 out of 448) of high School Degree holders have Years Of Experience of between 0 - 10 years. When we compare to the for Master's and PhD degree holders where just 61.8 percent and 22.1 percent have below 10 years Experience.

Finally, we see that Salary tend to increase across the age groups and by how long you stay at the company when we look at the average Salary by Age and average Salary by Years of Experience. And when we take a look at our comparison between age and Educational level, we see that similarly to our last comparison between Years of Experience and Educational Level, Most of the younger demographics have the lower qualifications as their highest form of education. We see that most Barchelor degree holders in the company are aged between 20 and 35 (about 85.77%) and which is true also for those who are high school degree holders (97.7%). However for the PhD and Master's Degree holders we see that 71.37% and 41.27% of their respective holders are above 35 years.